tests/percentage.test.cpp

```diff
@@ -53,6 +53,99 @@ boost::ut::suite percentage_test = []() {
       hal::percentage<double> assign_double2(assign_double);
       expect(that % assign_double.value() == assign_double2.value());
     };
+    "const equals zero"_test = []() {
+      // Setup
+      hal::percentage<float> value1 = hal::percentage<float>(0.0f);
+      hal::percentage<float> value2 = hal::percentage<float>(0.1f - 0.1f);
```

Comment:
NIT: Can replace this with auto.
NIT: Can also add constexpr to ensure that these are usable at compile time.
```suggestion
      auto value1 = hal::percentage<float>(0.0f);
      auto value2 = hal::percentage<float>(0.1f - 0.1f);
```

Ditto for the rest in the test.

Reply:
Add constexpr?